docker/native/README.md

```diff
@@ -0,0 +1,31 @@
+# GraalVM Based Native Apache Kafka Docker Image
+
+## Introduction
+- The Native Apache Kafka Docker Image can launch brokers with sub-second startup time and minimal memory footprint by leveraging native Kafka executable.
+- The native Kafka executable is built by compiling Apache Kafka code ahead-of-time using the [GraalVM native-image tool](https://www.graalvm.org/jdk21/reference-manual/native-image/).
+- This image is experimental and intended for local development and testing purposes only; it is not recommended for production use.
+- This is introduced with [KIP-974](https://cwiki.apache.org/confluence/display/KAFKA/KIP-974%3A+Docker+Image+for+GraalVM+based+Native+Kafka+Broker).
+
+## Native-Image reachability metadata
+The native-image tool performs static analysis while building a native binary to determine the dynamic features(the dynamic language features of the JVM, including reflection and resource handling, compute the dynamically-accessed program elements such as invoked methods or resource URLs at runtime), but it cannot always exhaustively predict all uses. 
+To ensure inclusion of these elements into the native binary, reachability metadata should be provided to the native-image builder. 
+These metadata configs are provided in the `native-image-configs` directory.
+
+## How to generate reachability metadata
+GraalVM provide an option to create these configs automatically by running the application normally with the [native-image agent](https://www.graalvm.org/latest/reference-manual/native-image/metadata/AutomaticMetadataCollection/) attached.
+The configs in the `native-image-configs` directory are generated by running existing Apache Kafka System Tests(using GraalVM JIT and attaching the native-image agent to the process) as they are quite exhaustive.
+
+## Limitations of Native Apache Kafka executable
+- Dynamic Features: For any new dynamic feature added or modified in Apache Kafka, the corresponding metadata configurations must be added or updated in the `native-image-configs`.
```

Comment:
I think this needs more details, to explain the steps to detect when the configs need to be updated and how the configs will be updated.
Is it something that RM needs to check before releasing the graalvm image? I think it would be good if we can clarify this.

Reply:
Currently there is no formulated steps for adding native configs. Initially we will be relying on the system tests results(failures) to catch if any configs are required. 
Note: It is not the responsibility of the RM to check for this.